chia/full_node/coin_store.py

```diff
@@ -63,8 +64,20 @@ async def create(cls, db_wrapper: DBWrapper2) -> CoinStore:
             log.info("DB: Creating index coin_spent_index")
             await conn.execute("CREATE INDEX IF NOT EXISTS coin_spent_index on coin_record(spent_index)")
 
-            log.info("DB: Creating index coin_puzzle_hash")
-            await conn.execute("CREATE INDEX IF NOT EXISTS coin_puzzle_hash on coin_record(puzzle_hash)")
+            # We're keeping coin_puzzle_hash index for existing DBs and adding
+            # coin_record_ph_si_idx for the new ones.
+            async with conn.execute(
+                "SELECT name FROM sqlite_master WHERE type = 'index' AND name = 'coin_puzzle_hash'"
+            ) as cursor:
+                no_coin_puzzle_hash_index = await cursor.fetchone() is None
+            if no_coin_puzzle_hash_index:
+                log.info("DB: Creating index coin_record_ph_si_idx")
+                await conn.execute(
+                    "CREATE INDEX IF NOT EXISTS coin_record_ph_si_idx ON coin_record(puzzle_hash, spent_index)"
```

Comment:
I really don't think we want an index on the `spent_index` itself. We want an index on whether it's zero or non-zero.